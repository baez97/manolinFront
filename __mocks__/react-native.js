function AsyncStorageClass() {
    this.setItem = jest.fn((key, value) => {
        return new Promise((resolve, reject) => {
            resolve();
        })
    });

    this.getItem = jest.fn((key, value) => {
        return new Promise((resolve, reject) => {
            resolve("THIS_IS_THE_SAVED_TOKEN");
        })
    });

    this.removeItem = jest.fn((key) => {
        return  new Promise((resolve, reject) => {
            resolve();
        })
    });

    this.removeItem.mockReturnValueOnce(new Promise((res,rej) => {
        rej({message: "THIS IS AN ERROR"});
    }))
}

module.exports.AsyncStorage = new AsyncStorageClass();