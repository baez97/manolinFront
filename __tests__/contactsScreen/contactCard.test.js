import React from 'react';
import { Linking }Â from 'react-native';
import renderer from 'react-test-renderer';
import ContactCard from '../../components/contactsComponents/contactCard';

jest.unmock("react-native")
jest.unmock('react')
jest.unmock("react-test-renderer")
jest.unmock('expo');

describe("ContactCard", () => {
    describe("Methods", () => {
        const props = {
            contact: {
                name:"CONTACT_NAME",
                phone:"CONTACT_PHONE"
            },

            onPressFn: jest.fn()
        }
        Linking.openURL = jest.fn();

        const c = new ContactCard(props);

        it("HandleLongPress calls to Linking", () => {
            c.handleLongPress();

            expect( Linking.openURL ).toBeCalled();
        });

        it("HandlePress calls to the props OnPressFn function", () => {
            c.handlePress();

            expect( props.onPressFn ).toBeCalled();
            expect( props.onPressFn ).toBeCalledWith(props.contact)
        });
    });

    describe("View", () => {
        it("Matches the Snapshot", () => {
            const contact = {
                name: "CONTACT_NAME",
                phone: "CONTACT_PHONE"
            };

            const view = <ContactCard contact={ contact }/>;
            const rendered = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        });
    });
});