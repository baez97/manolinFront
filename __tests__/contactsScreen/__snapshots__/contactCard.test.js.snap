// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard View Matches the Snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4294967295,
        4293848814,
      ]
    }
    endPoint={
      Array [
        0.65,
        1,
      ]
    }
    startPoint={
      Array [
        0.55,
        0,
      ]
    }
    style={
      Object {
        "borderRadius": 35.5,
        "elevation": 5,
        "flexDirection": "row",
        "height": 169.79999999999998,
        "justifyContent": "space-between",
        "marginTop": 17.5,
        "paddingRight": 17.5,
        "width": 641.25,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "space-evenly",
          "paddingBottom": 8.75,
          "paddingLeft": 52.5,
          "paddingTop": 8.75,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1b7128",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 38.95833442608516,
            "textAlign": "left",
          }
        }
      >
        CONTACT_NAME
      </Text>
      <Text
        style={
          Object {
            "color": "#2db842",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 31.87500089407465,
            "textAlign": "left",
          }
        }
      >
        CONTACT_PHONE
      </Text>
    </View>
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 177,
          "width": 177,
        }
      }
    />
  </ExponentLinearGradient>
</View>
`;
